### ISSUE TITLE
gradient of confidence

### ISSUE THREAD

**Issue by weitong8591** [zju3dv/LoFTR#241]
May I ask if the gradients can pass through the confidence? Because when I train it, the gradients can not propagate through the confidence, instead, only the confidence matrix can work for the propagation. So, it is wired that the indice selection of confidence matrix get out of the computation graph. Is there any way I can build the loss from confidence scores? Thanks!

**Comment by zehongs** [comment]
could you provide a code sample for the "indice selection of confidence matrix"?

**Comment by weitong8591** [comment]
Sure, I mean for data['mconf'], no gradient can propagate through it, right? I think it is wired because it is slicing the confidence matrix, is there anything I misunderstood? Thanks for your reply!
https://github.com/zju3dv/LoFTR/blob/b4ee7eb0359d0062e794c99f73e27639d7c7ac9f/src/loftr/utils/coarse_matching.py#L196

**Comment by yash0307** [comment]
To be more specific, how to backprop through mconf? As of now, it does not seem feasible.

**Comment by yash0307** [comment]
never mind, this issue has be resolved. Feel free to close this.
